### DISCUSSION TITLE
How To Rotate Frame Item  or  Image ? Thanks

### DISCUSSION THREAD

**Discussion by YokZhang** [fltk-rs/fltk-rs#1447] [Q&A] [answered]
hello 
I view all discussion found 2 function
   draw_text_angled(45, "Text", 59, 50);  =>  work well
   fltk::draw::rotate(45.); => Not Work

    let mut frm= Frame::new(0, 0, 100, 100, None);
    frm.clear_visible_focus();
    let image = image::PngImage::load(r"D:\img.png").unwrap();
    frm.set_image_scaled(Some(image));
    frm.draw(|d| {
        // I want to Rotate this Frame but Not Work
        fltk::draw::rotate(45.);
    });

I dont know  How to Rotate in Fltk-rs
thanks

**Comment by MoAlyousef** [comment, marked as answer]
The rotate function is like the dom canvas rotate or cairo_rotate. It’s part of the drawing api, it does not rotate a widget. 
For what you’re doing, you would need to rotate the image itself then use frm.set_image_scaled. 

To rotate the image you can use something like the image crate.

**Comment by YokZhang** [comment]
thanks for your advice
i learn a lot

  fltk = { version = "^1.3", features = ["fltk-bundled"] }
  image = "*"

    use fltk::{
    frame::Frame,
    prelude::{GroupExt, WidgetBase, WidgetExt, WindowExt},
    };
    use image::io::Reader as ImageReader;

    fn  main() {
    let img = ImageReader::open(r"D:\img.png")
        .unwrap()
        .decode()
        .unwrap();
    let imgRot90 = img.rotate90();
    imgRot90
        .save(r"D:\img90.jpg")
        .unwrap();
    let image =
        fltk::image::JpegImage::load(r"D:\img90.jpg").unwrap();

    let app = fltk::app::App::default().with_scheme(fltk::app::Scheme::Base);
    let mut win = fltk::window::Window::default()
        .with_size(900, 300)
        .center_screen();
    let mut frm = Frame::new(0, 0, 500, 500, None);
    frm.set_image_scaled(Some(image));

    win.end();
    win.show();
    app.run().unwrap();
    }

**Reply by MoAlyousef** [reply]
You shouldn’t need to save into jpeg first, you could directly draw imgRot90 or use it to construct an fltk RgbImage:
See the example here
https://github.com/fltk-rs/demos/blob/master/image/src/main.rs

**Reply by YokZhang** [reply]
Thank you for your kind help
It is better
keep learning fltk-rs
